test/framework/clusterresourceset_helpers.go

```diff
@@ -0,0 +1,142 @@
+	Expect(input.ClusterProxy).ToNot(BeNil(), "Invalid argument. input.ClusterProxy can't be nil when calling DiscoverClusterResourceSetAndWaitForSuccess")
+	Expect(input.Cluster).ToNot(BeNil(), "Invalid argument. input.Cluster can't be nil when calling DiscoverClusterResourceSetAndWaitForSuccess")
+
+	mgmtClient := input.ClusterProxy.GetClient()
+	fmt.Fprintln(GinkgoWriter, "Discovering cluster resource set resources")
+	clusterResourceSets := GetClusterResourceSets(ctx, GetClusterResourceSetsInput{
+		Lister:    mgmtClient,
+		Namespace: input.Cluster.Namespace,
+	})
+
+	Expect(clusterResourceSets).NotTo(BeEmpty())
+
+	for _, crs := range clusterResourceSets {
+		Expect(crs.Spec.ClusterSelector).NotTo(BeNil())
+
+		PatchClusterLabel(ctx, PatchClusterLabelInput{
+			ClusterProxy: input.ClusterProxy,
+			Cluster:      input.Cluster,
+			Labels:       crs.Spec.ClusterSelector.MatchLabels,
+		})
+
+		fmt.Fprintln(GinkgoWriter, "Waiting for ClusterResourceSet resources to be created in workload cluster")
+		WaitForClusterResourceSetToApplyResources(ctx, WaitForClusterResourceSetToApplyResourcesInput{
+			ClusterProxy:       input.ClusterProxy,
+			Cluster:            input.Cluster,
+			ClusterResourceSet: crs,
+		}, intervals...)
+
+	}
+
+}
+
+// WaitForClusterResourceSetToApplyResourcesInput is the input for WaitForClusterResourceSetToApplyResources.
+type WaitForClusterResourceSetToApplyResourcesInput struct {
+	ClusterProxy       ClusterProxy
+	Cluster            *clusterv1.Cluster
+	ClusterResourceSet *addonsv1.ClusterResourceSet
+}
+
+// WaitForClusterResourceSetToApplyResources wait until all ClusterResourceSet resources are created in the matching cluster.
+func WaitForClusterResourceSetToApplyResources(ctx context.Context, input WaitForClusterResourceSetToApplyResourcesInput, intervals ...interface{}) {
+	Expect(ctx).NotTo(BeNil(), "ctx is required for WaitForClusterResourceSetToApplyResources")
+	Expect(input.ClusterProxy).ToNot(BeNil(), "Invalid argument. input.ClusterProxy can't be nil when calling WaitForClusterResourceSetToApplyResources")
+	Expect(input.Cluster).ToNot(BeNil(), "Invalid argument. input.Cluster can't be nil when calling WaitForClusterResourceSetToApplyResources")
+	Expect(input.ClusterResourceSet).NotTo(BeNil(), "Invalid argument. input.ClusterResourceSet can't be nil when calling WaitForClusterResourceSetToApplyResources")
+
+	fmt.Fprintln(GinkgoWriter, "Waiting until the resource is created in the workload cluster")
+	Eventually(func() bool {
+		for _, resource := range input.ClusterResourceSet.Spec.Resources {
+			if resource.Kind == string(addonsv1.SecretClusterResourceSetResourceKind) {
+				secret := &corev1.Secret{}
+				err := input.ClusterProxy.GetWorkloadCluster(ctx, input.Cluster.Namespace, input.Cluster.Name).GetClient().Get(ctx, types.NamespacedName{Name: resource.Name, Namespace: input.ClusterResourceSet.Namespace}, secret)
```

Comment:
Is this checking to see if the workload cluster has a secret in the CRS's namespace whose name is the same as the CRS's resource name? That doesn't seem right?